lib/Backend/GlobOpt.cpp

```diff
@@ -4701,143 +4701,147 @@ GlobOpt::CollectMemOpInfo(IR::Instr *instr, Value *src1Val, Value *src2Val)
 
     if (loop->GetLoopFlags().isInterpreted && !loop->GetLoopFlags().memopMinCountReached)
     {
-        TRACE_MEMOP_VERBOSE(loop, instr, _u("minimum loop count not reached"))
+        TRACE_MEMOP_VERBOSE(loop, instrEnd, _u("minimum loop count not reached"))
         loop->doMemOp = false;
         return false;
     }
     Assert(loop->doMemOp);
 
     bool isIncr = true, isChangedByOne = false;
-    switch (instr->m_opcode)
+    FOREACH_INSTR_IN_RANGE(instr, instrBegin->m_next, instrEnd)
     {
```

Comment:
@meg-gupta I think only `IsInstrInvalidForMemop` should have to loop over the possibly new instructions. We won't insert new Add/Sub/Inc/Dec/LdElem/StElem instructions while executing `OptInstr`